client/src/components/Modal.jsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from "react";
+import "./Modal.css";
+import PropTypes from "prop-types";
+
+function Modal({ show, onClose, children }) {
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (show && event.target.classList.contains("modalOverlay")) {
+        onClose();
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [show, onClose]);
```

Comment:
Nice update! Can you double-check `onClose` is required in the dependency arry here? I think it's not.

Reply:
I checked again, it's required!